apps/src/lab2/views/dialogs/GenericPrompt.tsx

```diff
@@ -19,45 +21,66 @@ export type GenericPromptProps = Required<Pick<GenericDialogProps, 'title'>> & {
 type GenericPromptBodyProps = {
   placeholder?: string;
   prompt: string;
-  setPrompt: (newPrompt: string) => void;
+  handlePromptChange: (newPrompt: string) => void;
```

Comment:
Nomenclature thing that tripped me up a bit: I would expect `prompt` to be whatever static text we use to prompt the user (such as "Enter a file name"), but am I correct that `prompt` here is what the user typed? Would it make sense to rename this to `input` or something?